projects/services/public-api-service-typescript/test/public-api-service.spec.ts

```diff
@@ -39,46 +41,134 @@ const contractInput: types.CallContractInput = {
   sender: senderAddress
 };
 
+class FakeVirtualMachineService extends Service implements types.VirtualMachineServer {
+  async initialize(): Promise<void> {
+
+  }
+
+  async shutdown(): Promise<void> {
+
+  }
+
+  @Service.RPCMethod
+  callContract(rpc: types.CallContractContext): void {
+    expect(rpc.req.contractAddress).to.be.eql({ address: "contractAddress" });
+    expect(rpc.req.payload).to.be.eql("some-payload");
+
+    rpc.res = { resultJson: JSON.stringify("some-answer") };
+  }
+
+  processTransactionSet(rpc: types.ProcessTransactionSetContext): void {
+    throw new Error("Method not implemented.");
+  }
+}
+
+class FakeTransactionPool extends Service implements types.TransactionPoolServer {
+  @Service.RPCMethod
+  addNewPendingTransaction(rpc: types.AddNewPendingTransactionContext): void {
+    expect(rpc.req).to.be.eql({ transaction });
+  }
+
+  getAllPendingTransactions(rpc: types.GetAllPendingTransactionsContext): void {
+    throw new Error("Method not implemented.");
```

Comment:
we need to open a discussion about the code smell that is exposed by the need to implement-but-then-not-really-implement these methods